### PR TITLE
[docs] TNX user guide

### PR DESCRIPTION
**Pull request by tosterberg**
## Description ##

Add the user guid for TransformersNeuronX LMI containers

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: serving/docs/lmi_new/user_guides/tnx_user_guide.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+## Model Artifacts Structure
+
+LMI Transformers-NeuronX expect the model to be standard HuggingFace format for runtime compilation. For loading of compiled models, both Optimum compiled models and split-models with a separate `neff` cache (compiled models must be compiled with the same Neuron compiler version and model settings).
+The source of the model could be:
+
+- model_id string from huggingface
+- s3 url that store the artifact that follows the HuggingFace model repo structure, the Optimum compiled model repo structure, or a split model with a second url for the `neff` cache.
+- A local path that contains everything in a folder follows HuggingFace model repo structure, the Optimum compiled model repo structure, or a split model with a second directory for the `neff` cache.
+
+### Standard HuggingFace model artifacts
```
Comment (lanking520) [inline serving/docs/lmi_new/user_guides/tnx_user_guide.md:12]: please move model artifacts description to that description doc @siddvenk created

Comment (tosterberg) [reply serving/docs/lmi_new/user_guides/tnx_user_guide.md]: Moved to the new location

File: serving/docs/lmi_new/deployment_guide/model-artifacts.md [hunk 2]
```diff
@@ -57,6 +57,76 @@ Please remember to turn on `option.trust_remote_code=true` or `OPTION_TRUST_REMO
         └── tokenizer.model
 ```
 
+## Neuron Pretrained Model Formats
+
+For pretrained models that will be compiled at runtime, the HuggingFace Transformers pretrained format is preferred.
+
+Model compile time can quickly become and issue for larger models, so compiled models are accepted in the following formats.
```
Comment (siddvenk) [inline serving/docs/lmi_new/deployment_guide/model-artifacts.md:64]: `and` -> `an`

File: serving/docs/lmi_new/user_guides/tnx_user_guide.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+We will add more model support for the future versions to have them tested. Please feel free to [file us an issue](https://github.com/deepjavalibrary/djl-serving/issues/new/choose) for more model coverage in CI.
+
+### Quantization
+
+Currently, we allow customer to use `option.quantize=static_int8` or `OPTION_QUANTIZE=static_int8` to load the model using `int8` weight quantization.
+
+## Quick Start Configurations
+
+Most of the LMI Transformers-NeuronX models use the following template:
```
Comment (siddvenk) [inline serving/docs/lmi_new/user_guides/tnx_user_guide.md:45]: What cases would not?

Comment (tosterberg) [reply serving/docs/lmi_new/user_guides/tnx_user_guide.md]: Added clarifying comments and link to the advanced configuration doc.

File: serving/docs/lmi_new/user_guides/tnx_user_guide.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+The source of the model could be:
+
+- model_id string from huggingface
+- s3 url that store the artifact that follows the HuggingFace model repo structure, the Optimum compiled model repo structure, or a split model with a second url for the `neff` cache.
+- A local path that contains everything in a folder follows HuggingFace model repo structure, the Optimum compiled model repo structure, or a split model with a second directory for the `neff` cache.
+
+## Supported Model architecture
+
+LMI is shipping vLLM 0.2.7 with 0.26.0 containers, 
+so technically we support all LLM that [vLLM 0.2.7 support](https://github.com/vllm-project/vllm/tree/v0.2.7?tab=readme-ov-file#about).
```
[the remark this answers is not in the document]
Comment (tosterberg) [reply serving/docs/lmi_new/user_guides/tnx_user_guide.md]: Removed

File: serving/docs/lmi_new/deployment_guide/model-artifacts.md (hunk 2 above)
Comment (tosterberg) [reply serving/docs/lmi_new/deployment_guide/model-artifacts.md]: Fixed - thanks!

File: serving/docs/lmi_new/user_guides/tnx_user_guide.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
+# Transformers-NeuronX Engine in LMI
+
+## Model Artifacts Structure
```
Comment (siddvenk) [inline serving/docs/lmi_new/user_guides/tnx_user_guide.md:3]: Can we add a link back to the model artifact tnx section somewhere in this section? lgtm after we add that

Comment (tosterberg) [reply serving/docs/lmi_new/user_guides/tnx_user_guide.md:3]: Updated

**Review by siddvenk — APPROVED** [review]

**Pull request merged**